include/emds/emds.h

```diff
@@ -116,6 +116,15 @@ int emds_entry_add(struct emds_dynamic_entry *entry);
  * completed, the data should not be changed, and the device should be halted.
  * This is a time-critical operation and will be processed as fast as possible.
  *
+ * When writing to flash it bypasses the flash driver. Therefore, when running
+ * with MPSL, make sure to uninitialize MPSL by calling @ref mspl_lib_uninit
+ * before calling this function. Otherwise, an assertion may be triggered by
+ * the exit of the function.
+ *
+ * Once called, it locks all interrupts until the write is finished.
+ *
+ * Must be called from an interrupt context.
```

Comment:
We need to stop mpsl before storing emergency data, otherwise long flash write will trigger an assertion in mpsl. To stop mpsl we need to call `mpsl_uninit` (or `mpsl_lib_uninit` in nRF Connect SDK 2.1.99). As per my understanding it is only safe to call `mpsl_uninit` from ISR. This creates a dependency for `emds_store`. Also, I'm thinking now, we should do it from the highest priority interrupt so that while going from `mpsl_uninit` to `emds_store` no interrupts with highest priority preempt this interrupt.

If you are not using mpsl, it should be safe to call `emds_store` from any context.